firefox-ios/firefox-ios-tests/Tests/ClientTests/TelemetryWrapperTests.swift

```diff
@@ -832,6 +832,53 @@ class TelemetryWrapperTests: XCTestCase {
         testEventMetricRecordingSuccess(metric: GleanMetrics.Awesomebar.searchResultTap)
     }
 
+    func test_AwesomebarImpressions_GleanIsCalled() {
+        let groupsKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.groups.rawValue
+        let groups = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Groups.adaptiveHistory.rawValue
+
+        let interactionKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.interaction.rawValue
+        let interaction = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Interaction.persistedSearchTerms.rawValue
+
+        let nCharsKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.nChars.rawValue
+        let nChars: Int32 = 5
+
+        let nResultsKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.nResults.rawValue
+        let nResults: Int32 = 12
+
+        let nWordsKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.nWords.rawValue
+        let nWords: Int32 = 1
+
+        let reasonKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.reason.rawValue
+        let reason = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Reason.pause.rawValue
+
+        let resultsKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.results.rawValue
+        let results = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.searchEngine.rawValue + ","
+                      + TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.tabToSearch.rawValue
+
+        let sapKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.sap.rawValue
+        let sap = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Sap.urlbar.rawValue
+
+        let searchModeKey = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.searchMode.rawValue
+        let searchMode = TelemetryWrapper.EventExtraKey.UrlbarTelemetry.SearchMode.bookmarks.rawValue
+
+        let extraDetails = [groupsKey: groups,
+                       interactionKey: interaction,
+                            nCharsKey: nChars,
+                          nResultsKey: nResults,
+                            nWordsKey: nWords,
+                            reasonKey: reason,
+                           resultsKey: results,
+                               sapKey: sap,
+                        searchModeKey: searchMode] as [String: Any]
```

Comment:
This is very strange looking spacing. Let's not do this, because a) it's not anywhere in the codebase, and b) because it's not a standard Swift way of doing things. For this kind of stuff, we either align items to the opening bracket, or we do a
```swift
let extra = [
    thing: one,
] as [String: Any]
```

Reply:
haha, this strange looking spacing was the Objective C style of aligning stuff by the operators